rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -129,12 +132,13 @@ void SequentialReader::open(
     topics[0].topic_metadata.serialization_format);
 }
 
-void SequentialReader::set_read_order(const rosbag2_storage::ReadOrder & order)
+bool SequentialReader::set_read_order(const rosbag2_storage::ReadOrder & order)
 {
-  if (storage_) {
-    storage_->set_read_order(order);
+  if (!storage_) {
```

Comment:
One might expect this to `return false` given the new API? I think that'll work OK - you have to set_read_order after open on the `Reader`s as well.

Reply:
If `storage_` is null at this point that indicates that open hasn't been called, which is programmer error rather than something we can derive read order support from.